android/jni/com/mapswithme/maps/bookmarks/data/Bookmark.cpp

```diff
@@ -81,4 +81,13 @@ Java_com_mapswithme_maps_bookmarks_data_Bookmark_nativeEncode2Ge0Url(
 {
   return jni::ToJavaString(env, frm()->CodeGe0url(getBookmark(cat, bmk), addName));
 }
+
+JNIEXPORT jstring JNICALL
+Java_com_mapswithme_maps_bookmarks_data_Bookmark_nativeGetAddress(
+     JNIEnv * env, jobject thiz, jint catId, jlong bmkId)
+{
+  Bookmark const * bmk = getBookmark(catId, bmkId);
+  search::AddressInfo const address = g_framework->NativeFramework()->GetAddressInfoAtPoint(MercatorBounds::FromLatLon(bmk->GetLatLon()));
```

Comment:
Ещё раз - адрес букмарок ВСЕГДА берётся этим методом. В конструкторе букмарки мы НЕ задаём адрес, чтобы он был легковесней. Потому что при конструировании множества букмарок при просмотре списка - нужно получать их быстро. А адрес там НЕ нужен.

По поводу неудачности - надо дорефакторить в ядре букмарки, тогда и здесь можно сделать красиво. Пропадали названия как раз потому, что я ошибочно посчитал, что названия букмарок уже можно брать из place_page::Info.
> On Mar 25, 2016, at 3:42 PM, Alex Zolotarev <notifications@github.com> wrote:
> 
> In android/jni/com/mapswithme/maps/bookmarks/data/Bookmark.cpp <https://github.com/mapsme/omim/pull/2526#discussion_r57440852>:
> 
> > @@ -81,4 +81,13 @@ Java_com_mapswithme_maps_bookmarks_data_Bookmark_nativeEncode2Ge0Url(
> >  {
> >    return jni::ToJavaString(env, frm()->CodeGe0url(getBookmark(cat, bmk), addName));
> >  }
> > +
> > +JNIEXPORT jstring JNICALL
> > +Java_com_mapswithme_maps_bookmarks_data_Bookmark_nativeGetAddress(
> > +     JNIEnv * env, jobject thiz, jint catId, jlong bmkId)
> > +{
> > +  Bookmark const * bmk = getBookmark(catId, bmkId);
> > +  search::AddressInfo const address = g_framework->NativeFramework()->GetAddressInfoAtPoint(MercatorBounds::FromLatLon(bmk->GetLatLon()));
> Казалось бы, при чем тут список букмарок к плейспейджу? Явно неудачная сейчас архитектура. Думаю, уже можно находить баги, когда при тычке в PP один пишется адрес, а когда там же поставишь букмарку, уже будет другой.
> … <x-msg://7/#>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> 
>  <https://github.com/mapsme/omim/pull/2526#discussion_r57440593> <https://github.com/mapsme/omim/pull/2526#discussion_r57440227> <https://github.com/mapsme/omim/pull/2526#discussion_r57439562> <https://github.com/mapsme/omim/pull/2526/files/c026c407dde3767df58339f1a8a905cbb630eeba#r57439562> <https://github.com/mapsme/omim/pull/2526/files/c026c407dde3767df58339f1a8a905cbb630eeba#r57440227> <https://github.com/mapsme/omim/pull/2526/files/c026c407dde3767df58339f1a8a905cbb630eeba#r57440593>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2526/files/c026c407dde3767df58339f1a8a905cbb630eeba#r57440852>